File: frontend/src/services/budgetService.js
/* ────────────────────────────────────────────
   budgetService — Budget tracking API calls
   ──────────────────────────────────────────── */

import api from './api';

const budgetService = {
  getBudgetItems: (tripId) =>
    api.get(`/api/trips/${tripId}/budgets`).then((r) => r.data),

  addBudgetItem: (tripId, data) =>
    api.post(`/api/trips/${tripId}/budgets`, data).then((r) => r.data),

  updateBudgetItem: (tripId, budgetId, data) =>
    api
      .put(`/api/trips/${tripId}/budgets/${budgetId}`, data)
      .then((r) => r.data),

  deleteBudgetItem: (tripId, budgetId) =>
    api.delete(`/api/trips/${tripId}/budgets/${budgetId}`),

  getBudgetSummary: (tripId) =>
    api.get(`/api/trips/${tripId}/budgets/summary`).then((r) => r.data),
};

export default budgetService;

File: backend/src/main/java/com/traveloop/service/interfaces/BudgetService.java
package com.traveloop.service.interfaces;

import com.traveloop.model.dto.request.BudgetRequest;
import com.traveloop.model.dto.response.BudgetResponse;
import com.traveloop.model.dto.response.BudgetSummaryResponse;

import java.util.List;
import java.util.UUID;

public interface BudgetService {

    BudgetResponse addBudgetItem(UUID tripId, UUID userId, BudgetRequest request);

    List<BudgetResponse> getBudgetItems(UUID tripId, UUID userId);

    BudgetSummaryResponse getBudgetSummary(UUID tripId, UUID userId);

    BudgetResponse updateBudgetItem(UUID tripId, UUID budgetId, UUID userId, BudgetRequest request);

    void deleteBudgetItem(UUID tripId, UUID budgetId, UUID userId);
}
